app/history.py

```diff
@@ -19,44 +19,65 @@
 import app.log
 import cPickle as pickle
 import os
+import hashlib
+import time
+import app.prefs
 
-data = {}
-path = None
-
-def get(keyPath, default=None):
-  global data
-  cursor = data
-  for i in keyPath[:-1]:
-    cursor = cursor.setdefault(i, {})
-  return cursor.get(keyPath[-1], default)
-
-def set(keyPath, value):
-  global data
-  cursor = data
-  for i in keyPath[:-1]:
-    cursor = cursor.setdefault(i, {})
-  cursor[keyPath[-1]] = value
-  #assert get(keyPath) == value
-
-def loadUserHistory(filePath):
-  global data, path
-  try:
-    path = filePath
-    if os.path.isfile(path):
-      with open(path, "rb") as file:
-        data = pickle.load(file)
-    else:
-      data = {}
-  except Exception, e:
-    app.log.error('exception')
-    data = {}
+fileHistory = {}
```

Comment:
Are all views and objects recreated when you open a new file? What if I made the fileHistory an attribute of the text buffer class? It's pretty much only used for initializing values upon loading anyway.